### ISSUE TITLE
Syslog full of "unknown container" warnings for minecraft server

### ISSUE THREAD

**Issue by chris42** [itzg/dockerfiles#189]
I get a lot of warnings while running the minecraft server:

`dockerd[645]: time="2018-01-22T12:43:15.454851239+01:00" level=warning msg="unknown container" container=65c5ea981f0bd1c136faf23b9d0ec9e8df51065ee8991783ecc701675c$`

Not sure if this is a docker problem, but I have this only with the minecraft server. All other images and containers work without this.
Any idea what this is?

**Comment by itzg** [comment]
Yes, it seems to be a problem at the Docker daemon level and probably just coincidental with the Minecraft server. With that said, the only idea I have is some tool (using any monitoring?) is trying to poll the status of a container that no longer exists.

**Comment by chris42** [comment]
It is actually the container id of the minecraft-server container, so the warning does not make sense. Will see if anyone in the docker forums knows anything about it, then update here.

**Comment by chris42** [comment]
I found the issue.

It is some wrong filtering introduced in 17.12.0-ce, see [https://github.com/moby/moby/issues/35891](https://github.com/moby/moby/issues/35891)
Fix is on the way.

Only interesting part is, that it occurres within the health check of the container on my system and the minecraft-server is doing a lot of health checks (multiple times per second) in comparison to other containers I have. Any idea on how to tune that down a bit?

**Comment by itzg** [comment]
Yes, the interval can be adjusted but the observed interval is strange since I’m just using the default, by not specifying an interval here

https://github.com/itzg/dockerfiles/blob/master/minecraft-server/Dockerfile#L20

I’ll look on some of my different systems to see where this happens.

**Comment by chris42** [comment]
It is a ping to localhost. Within my logs (below) I found the container trying to resolve some weird definition of localhost (_minecraft._tcp.localhost), with my network DNS. Which of course has no clue about the docker internals.
Maybe it just retries and that is why it happens so often? Other containers are doing the healthcheck every 30sec.

When deactivating the health check, the error and DNS query is gone.
Maybe the localhost definition is wrong? Might do some more testing tomorrow.

```
dockerd[645]: time="2018-01-22T22:30:35.425765103+01:00" level=debug msg="Service name To resolve: _minecraft._tcp.localhost."
dockerd[645]: time="2018-01-22T22:30:35.426050137+01:00" level=debug msg="Service name To resolve: _minecraft._tcp.localhost."
dockerd[645]: time="2018-01-22T22:30:35.426119253+01:00" level=debug msg="[resolver] query _minecraft._tcp.localhost. (SRV) from 172.19.0.2:34267, forwarding to udp:192.168.2.1"
dockerd[645]: time="2018-01-22T22:30:35.426669216+01:00" level=debug msg="[resolver] external DNS udp:192.168.2.1 did not return any SRV records for \"_minecraft._tcp.localhost.\""
```

**Comment by itzg** [comment]
Oh, good find. Hmm, maybe it’s something the mcstatus tool is doing, but then I didn’t see any references to that mDNS type name in their code.

**Comment by chris42** [comment]
I just tried to use 127.0.0.1 in the healthcheck, it still constructs a weird lookup and tries to send it to my resolver.
Not sure, but you might want to check the healthcheck on other systems, as I have no clue if the lookup is special to my setup or general.

For me the issue is solved, as
- General Fix for the Warning is coming: [moby Issue #35891](https://github.com/moby/moby/issues/35891)
- I can deactivate the healthcheck in my docker-compose (version 2.1 upwards):
```
healthcheck:
  disable: true
```

**Comment by itzg** [comment]
Thanks for all your investigation and info. Agreed, I’ll compare on some other systems.
